packages/twenty-front/src/modules/apollo/optimistic-effect/utils/triggerCreateRecordsOptimisticEffect.ts

```diff
@@ -15,15 +17,27 @@ export const triggerCreateRecordsOptimisticEffect = ({
   cache,
   objectMetadataItem,
   records,
+  getRelationMetadata,
 }: {
   cache: ApolloCache<unknown>;
   objectMetadataItem: ObjectMetadataItem;
   records: CachedObjectRecord[];
+  getRelationMetadata: ReturnType<typeof useGetRelationMetadata>;
```

Comment:
Shouldn't we turn trigger utils into hooks if we need to pass the return of another hook ?

Reply:
As discussed: I tried to turn the utils into hooks, but as `triggerDeleteRecordsOptimisticEffect` uses `triggerUpdateRelationsOptimisticEffect` and vice-versa, when turned into hooks, they cause the app to crash because of a cyclic dependency. Let's keep them as utils for now, we'll figure out later if there is a better way to write this.